### ISSUE TITLE
[FLI-111] Flakey Auth Cleanup Tests

### ISSUE THREAD

**Issue by markphelps** [flipt-io/flipt#1214]
```
=== RUN   TestAuthenticationStoreHarness/List_all_authentications_(3_per_page_ascending)
    testing.go:98: 
        	Error Trace:	/Users/markphelps/workspace/flipt/internal/storage/auth/sql/testing.go:98
        	Error:      	Not equal: 
        	            	expected: []*auth.Authentication{(*auth.Authentication)(0xc00019d620), (*auth.Authentication)(0xc00019d920), (*auth.Authentication)(0xc00019db00), (*auth.Authentication)(0xc00019dce0), (*auth.Authentication)(0xc00019dec0), (*auth.Authentication)(0xc0000e60c0), (*auth.Authentication)(0xc0000e62a0), (*auth.Authentication)(0xc0000e6480), (*auth.Authentication)(0xc0000e6660), (*auth.Authentication)(0xc0000e6840), (*auth.Authentication)(0xc0000e6a20), (*auth.Authentication)(0xc0000e6c00), (*auth.Authentication)(0xc0000e6de0), (*auth.Authentication)(0xc0000e6fc0), (*auth.Authentication)(0xc0000e71a0), (*auth.Authentication)(0xc0000e7380), (*auth.Authentication)(0xc0000e7560), (*auth.Authentication)(0xc0000e7740), (*auth.Authentication)(0xc0000e7920), (*auth.Authentication)(0xc0000e7b00), (*auth.Authentication)(0xc0000e7ce0), (*auth.Authentication)(0xc0000e7ec0), (*auth.Authentication)(0xc0000f60c0), (*auth.Authentication)(0xc0000f62a0), (*auth.Authentication)(0xc0000f6480), (*auth.Authentication)(0xc0000f6660), (*auth.Authentication)(0xc0000f6840), (*auth.Authentication)(0xc0000f6a20), (*auth.Authentication)(0xc0000f6c00), (*auth.Authentication)(0xc0000f6de0), (*auth.Authentication)(0xc0000f6fc0), (*auth.Authentication)(0xc0000f71a0), (*auth.Authentication)(0xc0000f7380), (*auth.Authentication)(0xc0000f7560), (*auth.Authentication)(0xc0000f7740), (*auth.Authentication)(0xc0000f7920), (*auth.Authentication)(0xc0000f7b00), (*auth.Authentication)(0xc0000f7ce0), (*auth.Authentication)(0xc0000f7ec0), (*auth.Authentication)(0xc00020a0c0), (*auth.Authentication)(0xc00020a2a0), (*auth.Authentication)(0xc00020a480), (*auth.Authentication)(0xc00020a660), (*auth.Authentication)(0xc00020a840), (*auth.Authentication)(0xc00020aa20), (*auth.Authentication)(0xc00020ac00), (*auth.Authentication)(0xc00020ade0), (*auth.Authentication)(0xc00020afc0), (*auth.Authentication)(0xc00020b1a0), (*auth.Authentication)(0xc00020b380), (*auth.Authentication)(0xc00020b560), (*auth.Authentication)(0xc00020b740), (*auth.Authentication)(0xc00020b920), (*auth.Authentication)(0xc00020bb00), (*auth.Authentication)(0xc00020bce0), (*auth.Authentication)(0xc00020bec0), (*auth.Authentication)(0xc0002740c0), (*auth.Authentication)(0xc0002742a0), (*auth.Authentication)(0xc000274480), (*auth.Authentication)(0xc000274660), (*auth.Authentication)(0xc000274840), (*auth.Authentication)(0xc000274a20), (*auth.Authentication)(0xc000274c00), (*auth.Authentication)(0xc000274de0), (*auth.Authentication)(0xc000274fc0), (*auth.Authentication)(0xc0002751a0), (*auth.Authentication)(0xc000275380), (*auth.Authentication)(0xc000275560), (*auth.Authentication)(0xc000275740), (*auth.Authentication)(0xc000275920), (*auth.Authentication)(0xc000275b00), (*auth.Authentication)(0xc000275ce0), (*auth.Authentication)(0xc000275ec0), (*auth.Authentication)(0xc0002e40c0), (*auth.Authentication)(0xc0002e42a0), (*auth.Authentication)(0xc0002e4480), (*auth.Authentication)(0xc0002e4660), (*auth.Authentication)(0xc0002e4840), (*auth.Authentication)(0xc0002e4a20), (*auth.Authentication)(0xc0002e4c00), (*auth.Authentication)(0xc0002e4de0), (*auth.Authentication)(0xc0002e4fc0), (*auth.Authentication)(0xc0002e51a0), (*auth.Authentication)(0xc0002e5380), (*auth.Authentication)(0xc0002e5560), (*auth.Authentication)(0xc0002e5740), (*auth.Authentication)(0xc0002e5920), (*auth.Authentication)(0xc0002e5b00), (*auth.Authentication)(0xc0002e5ce0), (*auth.Authentication)(0xc0002e5ec0), (*auth.Authentication)(0xc0003340c0), (*auth.Authentication)(0xc0003342a0), (*auth.Authentication)(0xc000334480), (*auth.Authentication)(0xc000334660), (*auth.Authentication)(0xc000334840), (*auth.Authentication)(0xc000334a20), (*auth.Authentication)(0xc000334c00), (*auth.Authentication)(0xc000334de0), (*auth.Authentication)(0xc000334fc0), (*auth.Authentication)(0xc0003351a0)}
        	            	actual  : []*auth.Authentication{(*auth.Authentication)(0xc000365da0), (*auth.Authentication)(0xc000365e60), (*auth.Authentication)(0xc000365f20), (*auth.Authentication)(0xc000376600), (*auth.Authentication)(0xc0003766c0), (*auth.Authentication)(0xc000376780), (*auth.Authentication)(0xc000376e40), (*auth.Authentication)(0xc000376f00), (*auth.Authentication)(0xc000376fc0), (*auth.Authentication)(0xc0003776e0), (*auth.Authentication)(0xc0003777a0), (*auth.Authentication)(0xc000377860), (*auth.Authentication)(0xc000377f20), (*auth.Authentication)(0xc000382000), (*auth.Authentication)(0xc0003820c0), (*auth.Authentication)(0xc000382780), (*auth.Authentication)(0xc000382840), (*auth.Authentication)(0xc000382900), (*auth.Authentication)(0xc000382fc0), (*auth.Authentication)(0xc000383080), (*auth.Authentication)(0xc000383140), (*auth.Authentication)(0xc000383800), (*auth.Authentication)(0xc0003838c0), (*auth.Authentication)(0xc000383980), (*auth.Authentication)(0xc00039c060), (*auth.Authentication)(0xc00039c120), (*auth.Authentication)(0xc00039c1e0), (*auth.Authentication)(0xc00039c8a0), (*auth.Authentication)(0xc00039c960), (*auth.Authentication)(0xc00039ca20), (*auth.Authentication)(0xc00039d0e0), (*auth.Authentication)(0xc00039d1a0), (*auth.Authentication)(0xc00039d260), (*auth.Authentication)(0xc00039d920), (*auth.Authentication)(0xc00039d9e0), (*auth.Authentication)(0xc00039daa0), (*auth.Authentication)(0xc0003aa180), (*auth.Authentication)(0xc0003aa240), (*auth.Authentication)(0xc0003aa300), (*auth.Authentication)(0xc0003aa9c0), (*auth.Authentication)(0xc0003aaa80), (*auth.Authentication)(0xc0003aab40), (*auth.Authentication)(0xc0003ab200), (*auth.Authentication)(0xc0003ab2c0), (*auth.Authentication)(0xc0003ab380), (*auth.Authentication)(0xc0003aba40), (*auth.Authentication)(0xc0003abb00), (*auth.Authentication)(0xc0003abbc0), (*auth.Authentication)(0xc0003be2a0), (*auth.Authentication)(0xc0003be360), (*auth.Authentication)(0xc0003be420), (*auth.Authentication)(0xc0003beae0), (*auth.Authentication)(0xc0003beba0), (*auth.Authentication)(0xc0003bec60), (*auth.Authentication)(0xc0003bf320), (*auth.Authentication)(0xc0003bf3e0), (*auth.Authentication)(0xc0003bf4a0), (*auth.Authentication)(0xc0003bfb60), (*auth.Authentication)(0xc0003bfc20), (*auth.Authentication)(0xc0003bfce0), (*auth.Authentication)(0xc0003d03c0), (*auth.Authentication)(0xc0003d0480), (*auth.Authentication)(0xc0003d0540), (*auth.Authentication)(0xc0003d0c00), (*auth.Authentication)(0xc0003d0cc0), (*auth.Authentication)(0xc0003d0d80), (*auth.Authentication)(0xc0003d1440), (*auth.Authentication)(0xc0003d1500), (*auth.Authentication)(0xc0003d15c0), (*auth.Authentication)(0xc0003d1c80), (*auth.Authentication)(0xc0003d1d40), (*auth.Authentication)(0xc0003d1e00), (*auth.Authentication)(0xc0003f04e0), (*auth.Authentication)(0xc0003f05a0), (*auth.Authentication)(0xc0003f0660)}
        	            	
```

^ when running locally

ex: when running in CI: [https://github.com/flipt-io/flipt/actions/runs/3644494900/jobs/6153800733](https://github.com/flipt-io/flipt/actions/runs/3644494900/jobs/6153800733)

Local logs attached

[flakey.txt](https://uploads.linear.app/d3f9d655-7ab0-49e9-ade8-b59504819c07/92a1edc2-bc7e-4be0-98ef-42d39f0c9fdc/bd9b20bb-0240-4b8c-9f92-eaffa4bdef6c)

<sub>From [SyncLinear.com](https://synclinear.com) | [FLI-111](https://linear.app/flipt/issue/FLI-111/flakey-auth-cleanup-tests)</sub>

**Comment by GeorgeMac** [comment]
I think we have two different tests here.

As you mentioned, there you have reproduced an `auth/sql` failure:
`TestAuthenticationStoreHarness/List_all_authentications_(3_per_page_ascending)`

Whereas, in the CI link, we have the `oplock/sql`  test suite failing:
`Test_Harness`

Output from Postgres logs:
```
2022/12/08 01:54:52 STDERR : 2022-12-08 01:54:52.063 UTC [139] ERROR:  canceling statement due to user request
```

This seems to relate to the issue I see happening infrequently. And why I turned on database container logging.

**Comment by GeorgeMac** [comment]
I believe this will address the `oplock/sql` flake: https://github.com/flipt-io/flipt/pull/1216/commits/ebfb1b948464f0d01d9581f329bb143a48c4acda

This is a normal operation behaviour. The test tries to obtain the lock with N goroutines concurrently competing over it.
At some point, we signal for them all to stop with a context cancel. And then we ignore context cancel errors.
Since they're valid errors as we stopped inflight processes. However, if its in the middle of a query then this can propagate from the postgres side as a canceled query error code. I've added support to the postgres error adaptor for this case.
